java/org/apache/tomcat/util/IntrospectionUtils.java

```diff
@@ -332,7 +341,15 @@ public static String replaceProperties(String value,
         }
         if (prev < value.length())
             sb.append(value.substring(prev));
-        return sb.toString();
+        String newval = sb.toString();
+        if (newval.indexOf('$') < 0) {
```

Comment:
Same, here: look for `${`, not `$`.

Reply:
Do we want to process "bla ${variable} bla" too?
If $ is not enough only ${...} makes sense no?

Reply:
I think the first variant is the way to go. We even have tests, that check, if `abc${normal}xyz` gets replaced.